### PR TITLE
[13_3_X] Remove unused variable

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

In 13_3_X IBs, the following warning is emitted by GCC:

```
<...>/src/DataFormats/PortableTestObjects/src/classes.cc:4:1:   required from here
  <...>/src/DataFormats/SoATemplate/interface/SoALayout.h:582:12: warning: unused variable 'size' [-Wunused-variable]
   582 |       auto size = onfile.metadata().size();                                                                            \
      |            ^~~~
<...>/src/DataFormats/SoATemplate/interface/SoALayout.h:582:12: note: in definition of macro 'GENERATE_SOA_LAYOUT'
  582 |       auto size = onfile.metadata().size();                                                                            \
      |            ^~~~
```

[full log](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el9_amd64_gcc12/CMSSW_13_3_X_2024-04-14-2300/DataFormats/PortableTestObjects)

See also `master` version of this function: https://github.com/cms-sw/cmssw/blob/master/DataFormats/SoATemplate/interface/SoALayout.h#L584-L588

#### PR validation:

Bot tests

### CODE REVIEW

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
@cms-sw/heterogeneous-l2 gentle ping

**Comment by fwyzard** [comment]
Let me rephrase: why are we still trying to fix warnings in 13.3.x IBs ?

**Comment by fwyzard** [comment]
Anyway, since it looks like @cms-sw/orp-l2 have no problems bypassing signatures when _somebody else_ asks, they can do it here.

**Comment by smuzaffar** [comment]
@fwyzard , this warning was not there and showed up a couple of weeks ago. Looks like some got merged in 13.3.X which triggered this warning. So better to fix it

**Comment by antoniovilela** [comment]
> Anyway, since it looks like @cms-sw/orp-l2 have no problems bypassing signatures when _somebody else_ asks, they can do it here.

what was the instance?

**Comment by antoniovilela** [comment]
+1

**Comment by antoniovilela** [comment]
merge

**Comment by fwyzard** [comment]
+heterogeneous

since it went in anyway...
